plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -149,14 +149,15 @@ private Optional<SystemTable> getRawSystemTable(SchemaTableName tableName, Conne
         if (!isHudiTable(session, tableOptional.get())) {
             return Optional.empty();
         }
-        switch (name.getTableType()) {
-            case DATA:
-                break;
-            case TIMELINE:
+        return switch (name.getTableType()) {
+            case DATA ->
+                // TODO remove DATA table type like https://github.com/trinodb/trino/commit/5dd017b4ebb668a63d42a45735d1caaaf566e179 for Iceberg and Delta
```

Comment:
Create a follow-up issue `good-first-issue` for this TODO task

Reply:
created https://github.com/trinodb/trino/issues/17973 but didn't make it gfi